source/3ds/gui_hard.c

```diff
@@ -1442,15 +1723,19 @@ static inline int handle_buttons(Button buttons[], int count) {
     // draw buttons
     for (int i = 0; i < count; i++) {
         if (buttons[i].hidden) continue;
-        u32 normal_colour = C2D_Color32(TINT_R, TINT_G, TINT_B, 255);
-        u32 pressed_colour = C2D_Color32(TINT_R*0.5, TINT_G*0.5, TINT_B*0.5, 255);
+        u32 normal_colour = TINT_COLOR;
+        u32 pressed_colour = TINT_50;
         C2D_DrawRectSolid(buttons[i].x, buttons[i].y, 0, buttons[i].w, buttons[i].h,
             pressed == i ? pressed_colour : normal_colour);
         if (selectedButton == &buttons[i]) {
-            u32 line_colour = C2D_Color32(0, 0, 0, 255);
-            float ls = 3.5;
-            float le = 5.5;
-            C2D_DrawRectSolid(buttons[i].x + 4, buttons[i].y + buttons[i].h - 4, 0, buttons[i].w - 8, 1, C2D_Color32(0, 0, 0, 255));
+            if (buttons[i].draw_selected_rect) {
+                C2D_DrawLine(buttons[i].x + 2, buttons[i].y + 2.5, BLACK, buttons[i].x + buttons[i].w - 2, buttons[i].y + 2.5, BLACK, 1, 0);
```

Comment:
Fun fact, to draw pixel perfect straight lines in citro2d, you have to add 0.5 to ONLY the unchanging dimension. I believe this also causes the line between the pause button and the touch screen buttons to get drawn incorrectly sometimes... 😕 (Also ask me how long _that_ took to discover 😛)